source/elements/oneDPL/source/sycl_kernels_api.rst

```diff
@@ -46,17 +48,112 @@ Supported functionality:
 
 Limitations:
 ------------
+
 The following deviations from the `C++ Standard`_ may apply:
 
 - ``random_device`` and ``seed_seq`` classes and related APIs in other classes are not required;
 - ``operator>>()``, ``operator<<()``, ``operator==()`` are not required;
 - specifying the size of a random number engine's state is not required;
 - distributions are only required to operate with floating point types applicable to supported SYCL devices.
 
+Extensions:
+-----------
+
+As an extension to the `C++ Standard`_, ``sycl::vec<Type, N>`` can be used as the data type template parameter for
+engines, engine adaptors, and distributions, where ``Type`` is one of data types supported by the corresponding
+class template in the standard. For such template instantiations, the ``result_type`` is also defined to
+``sycl::vec<Type, N>``.
+
+Engines, engine adaptors, and distributions additionally define ``scalar_type``, equivalent to the following:
+
+- ``using scalar_type = typename result_type::element_type;`` if ``result_type`` is ``sycl::vec<Type, N>``,
+- otherwise, ``using scalar_type = result_type;``
+
+The ``scalar_type`` is used instead of ``result_type`` in all contexts where a scalar data type is expected, including
+
+- the type of configuration parameters and properties,
+- the seed value type,
+- the input parameters of constructors,
+- the return value type of ``min()`` and ``max()`` methods, etc.
+
+Since ``scalar_type`` is the same as ``result_type`` except for template instantiations with ``sycl::vec``,
+class templates still meet the applicable requirements of the `C++ Standard`_.
+
+When instantiated with ``sycl::vec<Type,N>``, ``linear_congruential_engine`` and ``subtract_with_carry_engine`` may not
+formally satisfy the uniform random bit generator requirements defined by the `C++ Standard`_. Instead, the following
+alternative requirements apply: for an engine object ``g`` of type ``G``,
+
+- ``G::scalar_type`` is an unsigned integral type same as ``sycl::vec<Type,N>::element_type``,
+- ``G::min()`` and ``G::max()`` return a value of ``G::scalar_type``,
+- for each index ``i`` in the range [0, ``N``), ``G::min() <= g()[i]`` and ``g()[i] <= G::max()``.
+
+Effectively, these engines satisfy the standard uniform random bit generator requirements for each element
+of a ``sycl::vec`` returned by their ``operator()``.
+
+Similarly, for a distribution ``d`` of a type ``D`` that is a template instantiated with ``sycl::vec<Type,N>``,
+
+- ``D::scalar_type`` is the same as ``sycl::vec<Type,N>::element_type``,
+- ``D::min()`` and ``D::max()`` return a value of ``D::scalar_type``, and ``D::min() <= D::max()``,
+- ``operator()`` of a distribution returns a ``sycl::vec<Type,N>`` filled with random values
+  in the closed interval ``[D::min(), D::max()]``;
+
+The following engines and engine adaptors with predefined parameters are defined:
```

Comment:
```suggestion
The following template aliases for engines and engine adaptors are defined:
```

Reply:
The wording "with predefined parameters" is aligned with C++ std. Should we change it?

Reply:
No, I think we should keep it. From the RNG standpoint, it's exactly engines with predefined parameters.

Reply:
Kept it as it is now

Reply:
I disagree, because in the standard they don't have template aliases to the best of my knowledge. But I don't want to argue.

Reply:
https://eel.is/c++draft/rand.predef
```
using minstd_rand0 =
      linear_congruential_engine<uint_fast32_t, 16'807, 0, 2'147'483'647>;
```
and so on.